homeassistant/components/input_button/__init__.py

```diff
@@ -0,0 +1,171 @@
+    id_manager = collection.IDManager()
+
+    yaml_collection = collection.YamlCollection(
+        logging.getLogger(f"{__name__}.yaml_collection"), id_manager
+    )
+    collection.sync_entity_lifecycle(
+        hass, DOMAIN, DOMAIN, component, yaml_collection, InputButton.from_yaml
+    )
+
+    storage_collection = InputButtonStorageCollection(
+        Store(hass, STORAGE_VERSION, STORAGE_KEY),
+        logging.getLogger(f"{__name__}.storage_collection"),
+        id_manager,
+    )
+    collection.sync_entity_lifecycle(
+        hass, DOMAIN, DOMAIN, component, storage_collection, InputButton
+    )
+
+    await yaml_collection.async_load(
+        [{CONF_ID: id_, **(conf or {})} for id_, conf in config.get(DOMAIN, {}).items()]
+    )
+    await storage_collection.async_load()
+
+    collection.StorageCollectionWebsocket(
+        storage_collection, DOMAIN, DOMAIN, CREATE_FIELDS, UPDATE_FIELDS
+    ).async_setup(hass)
+
+    async def reload_service_handler(service_call: ServiceCall) -> None:
+        """Remove all input buttons and load new ones from config."""
+        conf = await component.async_prepare_reload(skip_reset=True)
+        if conf is None:
+            return
+        await yaml_collection.async_load(
+            [
+                {CONF_ID: id_, **(conf or {})}
+                for id_, conf in conf.get(DOMAIN, {}).items()
+            ]
+        )
+
+    homeassistant.helpers.service.async_register_admin_service(
+        hass,
+        DOMAIN,
+        SERVICE_RELOAD,
+        reload_service_handler,
+        schema=RELOAD_SERVICE_SCHEMA,
+    )
+
+    component.async_register_entity_service(SERVICE_PRESS, {}, "_async_press_action")
+
+    return True
+
+
+class InputButton(ButtonEntity, RestoreEntity):
+    """Representation of a button."""
+
+    _attr_should_poll = False
+
+    def __init__(self, config: ConfigType) -> None:
+        """Initialize a button."""
+        self._config = config
+        self.editable = True
+        self._attr_unique_id = config[CONF_ID]
+
+    @classmethod
+    def from_yaml(cls, config: ConfigType) -> ButtonEntity:
+        """Return entity instance initialized from yaml storage."""
+        button = cls(config)
+        button.entity_id = f"{DOMAIN}.{config[CONF_ID]}"
+        button.editable = False
+        return button
+
+    @property
+    def name(self) -> str | None:
+        """Return name of the button."""
+        return self._config.get(CONF_NAME)
+
+    @property
+    def icon(self) -> str | None:
+        """Return the icon to be used for this entity."""
+        return self._config.get(CONF_ICON)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, bool]:
+        """Return the state attributes of the entity."""
+        return {ATTR_EDITABLE: self.editable}
+
+    async def async_press(self) -> None:
+        """Press the button.
+
+        Left emtpty intentionally.
+        The input button itself doesn't trigger anything.
```

Comment:
Fine, but how is it used then?
An automation would trigger on a state change of the button, which is updated when it's pressed?
If that's how it should be used, will automations fire when the button's state is restored?

Reply:
> An automation would trigger on a state change of the button, which is updated when it's pressed?

Yep.

> If that's how it should be used, will automations fire when the button's state is restored?

That is not a concern, integrations are set up first (unless they timeout, which can't happen for input_button), before the automations are started.

Reply:
OK, makes sense.
And reloading an automation will also not fire, because the trigger would be setup to trigger on a state CHANGE?

Reply:
Added a test for that specific case in 78cda1b